File: Chronir-Android/core/designsystem/src/main/java/com/chronir/designsystem/molecules/SnoozeOptionButton.kt
package com.chronir.designsystem.molecules

import android.content.res.Configuration.UI_MODE_NIGHT_YES
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.chronir.designsystem.atoms.ChronirText
import com.chronir.designsystem.atoms.ChronirTextStyle
import com.chronir.designsystem.theme.ChronirTheme
import com.chronir.designsystem.tokens.ColorTokens
import com.chronir.designsystem.tokens.RadiusTokens
import com.chronir.designsystem.tokens.SpacingTokens

@Composable
fun SnoozeOptionButton(
    label: String,
    sublabel: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier
            .size(72.dp, 64.dp)
            .clip(RoundedCornerShape(RadiusTokens.Md))
            .background(ColorTokens.BackgroundTertiary)
            .clickable(onClick = onClick),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        ChronirText(text = label, style = ChronirTextStyle.HeadlineSmall)
        ChronirText(
            text = sublabel,
            style = ChronirTextStyle.LabelSmall,
            color = ColorTokens.TextSecondary
        )
    }
}

enum class SnoozeInterval {
    OneHour,
    OneDay,
    OneWeek
}

@Composable
fun SnoozeOptionBar(
    onSnooze: (SnoozeInterval) -> Unit,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier,
        horizontalArrangement = Arrangement.spacedBy(SpacingTokens.Medium)
    ) {
        SnoozeOptionButton("1", "hour", onClick = { onSnooze(SnoozeInterval.OneHour) })
        SnoozeOptionButton("1", "day", onClick = { onSnooze(SnoozeInterval.OneDay) })
        SnoozeOptionButton("1", "week", onClick = { onSnooze(SnoozeInterval.OneWeek) })
    }
}

@Preview(name = "Snooze Bar — Light", showBackground = true)
@Preview(name = "Snooze Bar — Dark", showBackground = true, uiMode = UI_MODE_NIGHT_YES)
@Composable
private fun SnoozeOptionBarPreview() {
    ChronirTheme(dynamicColor = false) {
        SnoozeOptionBar(
            onSnooze = {},
            modifier = Modifier.padding(SpacingTokens.Medium)
        )
    }
}

File: chronir/chronir/DesignSystem/Molecules/SnoozeOptionButton.swift
import SwiftUI

struct SnoozeOptionButton: View {
    let label: String
    let sublabel: String
    let action: () -> Void

    init(_ label: String, sublabel: String = "", action: @escaping () -> Void) {
        self.label = label
        self.sublabel = sublabel
        self.action = action
    }

    var body: some View {
        Button(action: action) {
            VStack(spacing: SpacingTokens.xxs) {
                ChronirText(label, style: .headlineSmall)
                if !sublabel.isEmpty {
                    ChronirText(sublabel, style: .labelSmall, color: ColorTokens.textSecondary)
                }
            }
            .frame(width: 72, height: 64)
            .background(ColorTokens.backgroundTertiary)
            .clipShape(RoundedRectangle(cornerRadius: RadiusTokens.md))
        }
    }
}

struct SnoozeOptionBar: View {
    let onSnooze: (SnoozeInterval) -> Void

    enum SnoozeInterval {
        case oneHour
        case oneDay
        case oneWeek
        case custom(TimeInterval)
    }

    var showCustomButton: Bool = false
    var onCustomTap: (() -> Void)?

    var body: some View {
        HStack(spacing: SpacingTokens.md) {
            SnoozeOptionButton("1", sublabel: "hour") { onSnooze(.oneHour) }
            SnoozeOptionButton("1", sublabel: "day") { onSnooze(.oneDay) }
            SnoozeOptionButton("1", sublabel: "week") { onSnooze(.oneWeek) }
            if showCustomButton {
                SnoozeOptionButton("...", sublabel: "custom") { onCustomTap?() }
            }
        }
    }
}

#Preview("Snooze Option Bar") {
    SnoozeOptionBar(onSnooze: { _ in })
        .padding()
        .background(ColorTokens.backgroundPrimary)
}

#Preview("Individual Button") {
    SnoozeOptionButton("5", sublabel: "min") {}
        .padding()
        .background(ColorTokens.backgroundPrimary)
}
